R/register_extendr.R

```diff
@@ -28,23 +28,23 @@
 #' @seealso [rextendr::document()]
 #' @export
 register_extendr <- function(path = ".", quiet = FALSE, force = FALSE, compile = NA) {
-  usethis_quiet <- getOption("usethis.quiet")
-  on.exit(options(usethis.quiet = usethis_quiet))
-  options(usethis.quiet = quiet)
+
+  if (quiet) {
+    withr::local_options(list("cli.default_handler" = function(...) { }))
+  }
 
   rextendr_setup(path = path)
 
   pkg_name <- pkg_name(path)
 
-  ui_i("Generating extendr wrapper functions for package: {.pkg {pkg_name}}.")
+  cli::cli_alert_info("Generating extendr wrapper functions for package: {.pkg {pkg_name}}.")
```

Comment:
Same, what's the difference from `cli::cli_inform()`? 
We do not really need a 1-to-1 match with the existing behavior, it is ok to improve it.